client/my-sites/plans/ecommerce-trial/upgrade-confirmation/index.tsx

```diff
@@ -1,22 +1,23 @@
 import { useTranslate } from 'i18n-calypso';
 import { useSelector } from 'react-redux';
+import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
 import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import Main from 'calypso/components/main';
 import BodySectionCssClass from 'calypso/layout/body-section-css-class';
 import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
 import { isRequestingSitePlans } from 'calypso/state/sites/plans/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import ConfirmationTask from './confirmation-task';
-import { getConfirmationTasks } from './confirmation-tasks';
+import { GetConfirmationTasks } from './confirmation-tasks';
 import type { AppState } from 'calypso/types';
 
 import './style.scss';
 
 const TrialUpgradeConfirmation = () => {
 	const selectedSite = useSelector( getSelectedSite );
 	const translate = useTranslate();
-
-	const tasks = getConfirmationTasks( { translate } );
+	const siteSlug = selectedSite?.slug;
+	const tasks = GetConfirmationTasks( { translate, siteSlug } );
```

Comment:
Can we pass the `siteId` in directly? That simplifies the logic in the function so it doesn't need to resolve the slug to a siteId when we already have the ID.
```suggestion
	const siteId = selectedSite?.ID;
	const tasks = GetConfirmationTasks( { translate, siteId } );
```